diff --git a/envpool/__init__.py b/envpool/__init__.py
index 1c46bb3..24e0c1d 100644
--- a/envpool/__init__.py
+++ b/envpool/__init__.py
@@ -24,7 +24,7 @@ from envpool.registration import (
   register,
 )
 
-__version__ = "0.8.4"
+__version__ = "0.8.4.1"
 __all__ = [
   "register",
   "make",
diff --git a/envpool/core/env.h b/envpool/core/env.h
index 28658e3..6ccf9bd 100644
--- a/envpool/core/env.h
+++ b/envpool/core/env.h
@@ -91,7 +91,7 @@ class Env {
       : max_num_players_(spec.config["max_num_players"_]),
         spec_(spec),
         env_id_(env_id),
-        seed_(spec.config["seed"_] + env_id),
+        seed_(spec.config["seed"_]),
         gen_(seed_),
         is_single_player_(max_num_players_ == 1),
         action_specs_(spec.action_spec.template AllValues<ShapeSpec>()),
diff --git a/setup.cfg b/setup.cfg
index 0480e60..3f58b72 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,6 @@
 [metadata]
 name = envpool
-version = 0.8.4
+version = 0.8.4.1
 author = "EnvPool Contributors"
 author_email = "sail@sea.com"
 description = "C++-based high-performance parallel environment execution engine (vectorized env) for general RL environments."
